src/tbox/utils/fnv32.h

```diff
@@ -0,0 +1,54 @@
+/*!The Treasure Box Library
+ * 
+ * TBox is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ * 
+ * TBox is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with TBox; 
+ * If not, see <a href="http://www.gnu.org/licenses/"> http://www.gnu.org/licenses/</a>
+ * 
+ * Copyright (C) 2016, Olexander Yermakov All rights reserved.
+ *
+ * @author      alexyer
+ * @file        fnv32.h
+ * @ingroup     utils
+ *
+ */
+#ifndef TB_UTILS_FNV32_H
+#define TB_UTILS_FNV32_H
+
+/* //////////////////////////////////////////////////////////////////////////////////////
+ * includes
+ */
+#include "prefix.h"
+
+/* //////////////////////////////////////////////////////////////////////////////////////
+ * extern
+ */
+__tb_extern_c_enter__
+
+/* //////////////////////////////////////////////////////////////////////////////////////
+ * interfaces
+ */
+
+/*! encode fnv32
+ *
+ * @param data      the data
+ *
+ * @return          the fnv32 value
+ */
+tb_uint32_t         tb_fnv32_encode(tb_byte_t const* data);
```

Comment:
It will be better if you can add an argument for the data size.

e.g. 
tb_uint32_t tb_fnv32_encode(tb_byte_t const* data, tb_size_t size);

Thanks! : )